### DISCUSSION TITLE
add vite plugin to quasar config

### DISCUSSION THREAD

**Discussion by abdo-spices** [quasarframework/quasar#17270] [General - Components / Directives / etc] [answered]
i have quasar config 
i was trying to to add this following vite config 

```js
import Vue from '@vitejs/plugin-vue'
import Markdown from 'vite-plugin-md'

export default {
  plugins: [
    Vue({
      include: [/\.vue$/, /\.md$/], // <--
    }),
    Markdown(),
  ],
}
```

```js
      vitePlugins: [
        [
          "@vitejs/plugin-vue",
          {
            include: [/\.vue$/, /\.md$/],
          },
        ],
        ["vite-plugin-md"],
        [
          "@intlify/vite-plugin-vue-i18n",
          {
            include: path.resolve(__dirname, "./src/i18n/**"),
          },
        ],
      ],
```
but it is not working
how i can import my own vite config inside quasar or call the plugins in the right way

**Comment by yusufkandemir** [comment, marked as answer]
Quasar already includes `@vitejs/plugin-vue`, so you shouldn't add it there. Use `quasar.config file > build > viteVuePluginOptions` to pass customized options to it.
https://quasar.dev/quasar-cli-vite/handling-vite#vite-vue-plugin-options

Here is an example from Quasar Docs source code:
https://github.com/quasarframework/quasar/blob/fe421fbd7dd183fd0b223cb2bdd8b66e86becc96/docs/quasar.config.js#L32-L34

**Reply by abdo-spices** [reply]
what about vite-plugin-md

**Reply by yusufkandemir** [reply]
Did you try this and it still didn't work, or are you asking hypothetically?

**Reply by AhmedHamdy-B** [reply]
I have these configs in my Quasar project:
![image](https://github.com/quasarframework/quasar/assets/56585994/4cf606ee-e7ef-4917-a179-4d7f994d88b9)
![image](https://github.com/quasarframework/quasar/assets/56585994/631ad62c-3ccb-49a9-8fdd-4db941124196)
and it shows this error:
![image](https://github.com/quasarframework/quasar/assets/56585994/d2e65bb2-5810-418c-b2bd-4fe77d5ea971)
Note:
I am using these versions in my package.json
![image](https://github.com/quasarframework/quasar/assets/56585994/5bd95ad5-e538-45e9-8c27-a4649e0652f5)

**Reply by AhmedHamdy-B** [reply]
@yusufkandemir we are working on the same issue together, so this code is regarding the discussion, thanks in advance ♥

**Reply by yusufkandemir** [reply]
https://github.com/antfu/vite-plugin-md/issues/127
It's an ongoing issue regarding vite-plugin-md. From what I can see in the comments, a common workaround is using ESM. To do that, you should upgrade to app-vite v2 beta. Otherwise, the issue needs to be addressed in vite-plugin-md.
